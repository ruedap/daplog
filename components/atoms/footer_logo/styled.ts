import styled from 'styled-components'
import Styles from '@/styles'

export const Img = styled.img`
  opacity: ${Styles.funcs.fibo('md', 'alpha')};
  transition: opacity 0.3s;
  &:hover {
    opacity: ${Styles.funcs.fibo('lg', 'alpha')};
  }
`
