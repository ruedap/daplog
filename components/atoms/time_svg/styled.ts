import styled from 'styled-components'
import Styles from '@/styles'

export const Svg = styled.svg`
  display: block;
  left: 0;
  position: absolute;
  top: 0;
`

export const Circle = styled.circle`
  fill: ${({ theme }) => theme.colors.key[3]};
`

export const Text = styled.text`
  ${Styles.mixins.fontSmoothing()};
  fill: #fff;
  font-family: var(--b-fontFamily-fjalla);
  font-size: ${Styles.funcs.fibo('xs', 'rem')};
  letter-spacing: 0.1em;
`
