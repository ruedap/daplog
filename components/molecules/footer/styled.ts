import styled, { css } from 'styled-components'
import Styles from '@/styles'

export const Root = styled.footer`
  display: block;
  position: relative;
`

export const Container = styled.div`
  ${({ theme }) => css`
    ${Styles.mixins.container};
    background-image: linear-gradient(
      ${theme.colors.key[3]} 0,
      ${theme.colors.key[4]} 100%
    );

    ${theme.mq.up.lg} {
      &::before {
        ${Styles.mixins.dapicons};
        color: ${theme.colors.key[2]};
        content: "ruedap";
        font-size: ${theme.fontSizes.md.rem};
        left: 50%;
        position: absolute;
        top: 377px;
        transform: translate(-50%, -100%);
      }
    }
  `}
`

export const Heading = styled.div`
  margin: 0;
  position: relative;
`

export const LogoSpacer = styled.div`
  display: block;
  font-size: 0;
  /* 1129 / 987 = 1.143870314 */
  padding-bottom: 114.387%;
  position: relative;
  width: 100%;
`

export const LogoLink = styled.a`
  bottom: 0;
  display: block;
  height: auto;
  left: 0;
  margin: auto;
  position: absolute;
  right: 0;
  top: 0;
`
