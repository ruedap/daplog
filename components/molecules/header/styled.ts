import styled from 'styled-components'
import Styles from '@/styles'

export const Inner = styled.div`
  ${Styles.mixins.container}
  background-color: #fff;
`

export const Heading = styled.div`
  margin: 0;
  padding-bottom: 11.854%;
`

export const LogoLink = styled.a`
  display: block;
  position: relative;
  width: 100%;
`

export const LogoSpacer = styled.div`
  display: block;
  font-size: 0;
  margin-top: -50%;
  padding-bottom: 100%;
  position: relative;
  width: 100%;
`

export const LogoOuter = styled.div`
  background-color: var(--b-bgColor-body);
  background-image: radial-gradient(var(--b-bgColor-body), var(--b-color-key3));
  border-radius: 100%;
  bottom: 0;
  display: block;
  height: auto;
  left: 0;
  margin: auto;
  position: absolute;
  right: 0;
  top: 0;
  width: 100%;
`
