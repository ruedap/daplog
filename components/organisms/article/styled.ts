import styled, { css } from 'styled-components'
import Styles from '@/styles'
import * as StyledMarkdown from './_styled_markdown'

export const TimeContainer = styled.div`
  display: block;
  left: 0;
  margin: auto;
  max-width: var(--b-maxWidth-container);
  pointer-events: none;
  position: absolute;
  right: 0;
  top: 0;
  width: 100%;
`

export const TimeSpacer = styled.div`
  display: block;
  font-size: 0;
  padding-bottom: 61.8%;
  position: relative;
  width: 100%;
`

export const Title = styled.h1`
  ${({ theme }) => css`
    ${Styles.mixins.fontSmoothing()}
    font-family: var(--b-fontFamily-crimson);
    font-size: ${Styles.funcs.fibo('xs', 'rem')};
    font-weight: normal;
    line-height: 1.618;
    margin-top: 0;
    margin-bottom: ${Styles.funcs.fibo('sm', 'px')};
    margin-left: auto;
    margin-right: auto;
    max-width: 877px;
    text-align: center;
    width: 90%;

    ${theme.mq.up.md} {
      font-size: ${Styles.funcs.fibo('sm', 'rem')};
      margin-bottom: ${Styles.funcs.fibo('md', 'px')};

      span::after {
        content: "\\000a";
        white-space: pre-line;
      }
    }
  `}
`

export const GitHubLink = styled.a`
  ${({ theme }) => css`
    ${Styles.mixins.articleSectionMark('xs')}
    height: ${Styles.funcs.fibo('xs', 'px')};
    margin-bottom: calc(${Styles.funcs.fibo('sm', 'px')} + 10px);
    width: calc(${Styles.funcs.fibo('xs', 'px')} * 2);

    ${theme.mq.up.md} {
      ${Styles.mixins.articleSectionMark('sm')}
      height: ${Styles.funcs.fibo('sm', 'px')};
      margin-bottom: calc(${Styles.funcs.fibo('md', 'px')} + 10px);
      width: calc(${Styles.funcs.fibo('sm', 'px')} * 2);
    }
  `}
`

export const Body = styled.section`
  margin: 0 auto;
  max-width: 699px;
  width: 90%;
  ${StyledMarkdown.base}
  ${StyledMarkdown.heading}
  ${StyledMarkdown.p}
  ${StyledMarkdown.img}
  ${StyledMarkdown.strong}
  ${StyledMarkdown.list}
  ${StyledMarkdown.hr}
  ${StyledMarkdown.figure}
  ${StyledMarkdown.cite}
  ${StyledMarkdown.blockquote}
  ${StyledMarkdown.small}
  ${StyledMarkdown.footnotes}
  ${StyledMarkdown.highlight}
  ${StyledMarkdown.code}
  ${StyledMarkdown.table}
  /* extra styles */
  ${StyledMarkdown.bookCover}
  ${StyledMarkdown.embed}
  ${StyledMarkdown.dapicons}
`

export const Nav = styled.nav`
  ${({ theme }) => css`
    margin-top: ${Styles.funcs.fibo('xl', 'px')};
    padding-bottom: ${Styles.funcs.fibo('xl', 'px')};

    ${theme.mq.up.md} {
      margin-top: ${Styles.funcs.fibo('xl2', 'px')};
      padding-bottom: ${Styles.funcs.fibo('xl2', 'px')};
    }
  `}
`

export const NavLink = styled.a`
  ${({ theme }) => css`
    ${Styles.mixins.linkColors('#fff', '#fff', theme.colors.key[4], theme.colors.key[4], theme.colors.key[5])}
    ${Styles.mixins.fontSmoothing()}
    ${Styles.mixins.dapicons}
    background-color: ${theme.colors.key[2]};
    border-radius: 100%;
    display: block;
    font-size: 2.6rem;
    width: ${Styles.funcs.fibo('md', 'px')};
    height: ${Styles.funcs.fibo('md', 'px')};
    line-height: ${Styles.funcs.fibo('md', 'px')};
    margin: auto;
    text-align: center;
    text-decoration: none;
  `}
`
