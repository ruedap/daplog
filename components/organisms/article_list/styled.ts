import styled, { css } from 'styled-components'
import Styles from '@/styles'

export const Root = styled.ul`
  margin-top: 0;
  margin-bottom: 0;
  padding-left: 0;
  list-style: none;
  color: #fff;
`

export const Item = styled.li`
  ${Styles.mixins.fontSmoothing()}
  font-size: 1.4rem;
  line-height: ${Styles.funcs.fibo('sm', 'px')};
  margin-bottom: 1px;

  ${Styles.mq.up.md(css`
    font-size: 1.8rem;
    line-height: ${Styles.funcs.fibo('md', 'px')};
  `)}
`

export const YearHeading = styled(Item)`
  ${({ theme }) => css`
    background-color: ${theme.colors.key[3]};
    color: ${theme.colors.key[5]};
    font-family: var(--b-fontFamily-fjalla);
    height: ${Styles.funcs.fibo('sm', 'px')};
    letter-spacing: 0.1em;
    text-align: center;

    ${Styles.mq.up.md(css`
      height: ${Styles.funcs.fibo('md', 'px')};
    `)}
  `}
`

export const Year = styled.span`
  display: none;
`
export const Dot = Year
