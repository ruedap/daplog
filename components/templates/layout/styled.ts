import styled from 'styled-components'
import Styles from '@/styles'

export const Container = styled.div`
  ${Styles.mixins.container}
  background-color: ${({ theme }) => theme.colors.bg.content};
`
