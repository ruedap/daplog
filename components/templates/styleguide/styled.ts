export * from './_styled_tag4'

import styled, { css, keyframes, Keyframes } from 'styled-components'

export const Title = styled.h1`
  font-size: 4rem;
  margin: 0;
  color: ${({ theme }) => theme.colors.primary};
`

const base = css`
  background-color: white;
  color: cornflowerblue;
  border: 1px solid lightgreen;
  border-right: none;
  border-bottom: none;
  box-shadow: 5px 5px 0 0 lightgreen, 10px 10px 0 0 lightyellow;
  transition: all 0.1s linear;
  margin: 0 3rem 3rem;
  padding: 1rem 0.5rem;
`

const hover = css`
  &:hover {
    color: ${({theme}) => theme.colors.primary};
    border-color: aqua;
    box-shadow: -15px -15px 0 0 aqua, -30px -30px 0 0 cornflowerblue;
  }
`

export const prop = css`
  font-size: 200%;
  color: indigo;
`

export const bounce = keyframes`
  from {
    transform: scale(1.01);
  }
  to {
    transform: scale(0.99);
  }
`

export const shake = keyframes`
  10%, 90% { transform: translateY(-1px); }
  20%, 80% { transform: translateY(2px); }
  30%, 50%, 70% { transform: translateY(-4px); }
  40%, 60% { transform: translateY(4px); }
`

export const Tag1 = styled.div`
  ${base};
`

export const Tag2 = styled.div`
  ${base};
  ${hover};
  & code {
    background-color: linen;
  }
`

export const Tag3 = styled.div<{ animation: Keyframes }>`
  ${base};
  ${hover};
  & code {
    background-color: linen;
  }
  animation: ${({ animation }) => animation } 0.2s infinite ease-in-out alternate;
`
