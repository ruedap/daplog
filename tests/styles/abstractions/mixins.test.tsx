import styled, { css } from 'styled-components'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import {
  fontSmoothing,
  linkColors,
  ellipsis,
  articleSectionMark
} from '@/styles/abstracts/mixins'
import { lightTheme } from '@/styles/theme'
import { renderWithTheme } from '@/tests/helpers'

// const stripIndent = (str) => str.replace(/^ +/gm, '').replace(/\r?\n/g, '')

describe('fontSmoothing', () => {
  test('should return same styles.', () => {
    const actual = fontSmoothing().join('')
    const expected = css`
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
`.join('')
    expect(actual).toEqual(expected)
  })

  test('should return same styles.', () => {
    const actual = fontSmoothing(false).join('')
    const expected = css`
  -webkit-font-smoothing: subpixel-antialiased;
  -moz-osx-font-smoothing: auto;
`.join('')
    expect(actual).toEqual(expected)
  })
})

describe('linkColors', () => {
  test('should return same styles.', () => {
    const actual = linkColors().join('')
    const expected = css``.join('')
    expect(actual).toEqual(expected)
  })

  test('should return same styles.', () => {
    const Styled = styled.div`${linkColors('#336699')}`
    const tree = renderer.create(<Styled />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should return same styles.', () => {
    const Styled = styled.div`${linkColors('#336699', 'black')}`
    const tree = renderer.create(<Styled />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should return same styles.', () => {
    const Styled = styled.div`${linkColors('#336699', 'black', undefined, 'pink')}`
    const tree = renderer.create(<Styled />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should return same styles.', () => {
    const color = lightTheme.colors.key[3]
    const Styled = styled.div`${linkColors(
      '#336699', 'black', color, 'pink', '#000'
    )}`
    const tree = renderer.create(<Styled />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('ellipsis', () => {
  test('should return same styles.', () => {
    const actual = ellipsis(2).join('')
    const arg = 2
    const expected = css`
  @supports (-webkit-line-clamp: ${arg}) {
    display: -webkit-box;
    overflow: hidden;
    -webkit-line-clamp: ${arg};
    -webkit-box-orient: vertical;
  }
  @supports not (-webkit-line-clamp: ${arg}) {
    display: block;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
  }
`.join('')
    expect(actual).toEqual(expected)
  })
})

describe('articleSectionMark', () => {
  test('should return same styles.', () => {
    const Styled = styled.div`${articleSectionMark('sm')}`
    const tree = renderWithTheme(<Styled />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should return same styles.', () => {
    const styles = css`
      color: ${({ theme }) => theme.colors.text.body};
      margin-top: 20px;
      font-size: 13px;
    `
    const Styled = styled.div`${articleSectionMark('md', styles)}`
    const tree = renderWithTheme(<Styled />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
