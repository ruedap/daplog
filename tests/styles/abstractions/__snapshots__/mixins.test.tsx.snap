// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`articleSectionMark should return same styles. 1`] = `
.c0 {
  display: block;
  margin-left: auto;
  margin-right: auto;
  position: relative;
}

.c0::before,
.c0::after {
  background-color: rgba(24,63,83,0.21);
  border-radius: 100%;
  content: '';
  display: block;
  height: 34px;
  left: 50%;
  margin: 0;
  padding: 0;
  position: absolute;
  top: 0;
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
  -webkit-transform-origin: center center;
  -ms-transform-origin: center center;
  transform-origin: center center;
  width: 34px;
}

.c0::before {
  margin-left: 11.333333333333334px;
}

.c0::after {
  margin-left: -11.333333333333334px;
}

<div
  className="c0"
/>
`;

exports[`articleSectionMark should return same styles. 2`] = `
.c0 {
  display: block;
  margin-left: auto;
  margin-right: auto;
  position: relative;
  color: #21272d;
  margin-top: 20px;
  font-size: 13px;
}

.c0::before,
.c0::after {
  background-color: rgba(24,63,83,0.21);
  border-radius: 100%;
  content: '';
  display: block;
  height: 55px;
  left: 50%;
  margin: 0;
  padding: 0;
  position: absolute;
  top: 0;
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
  -webkit-transform-origin: center center;
  -ms-transform-origin: center center;
  transform-origin: center center;
  width: 55px;
}

.c0::before {
  margin-left: 18.333333333333332px;
}

.c0::after {
  margin-left: -18.333333333333332px;
}

<div
  className="c0"
/>
`;

exports[`linkColors should return same styles. 1`] = `
.c0 {
  color: #336699;
}

<div
  className="c0"
/>
`;

exports[`linkColors should return same styles. 2`] = `
.c0 {
  color: #336699;
}

.c0:visited {
  color: black;
}

<div
  className="c0"
/>
`;

exports[`linkColors should return same styles. 3`] = `
.c0 {
  color: #336699;
}

.c0:visited {
  color: black;
}

.c0:hover {
  color: pink;
}

<div
  className="c0"
/>
`;

exports[`linkColors should return same styles. 4`] = `
.c0 {
  color: #336699;
}

.c0:visited {
  color: black;
}

.c0:focus {
  color: rgba(24,63,83,0.34);
}

.c0:hover {
  color: pink;
}

.c0:active {
  color: #000;
}

<div
  className="c0"
/>
`;
